using/solving-exercises/tdd.md

```diff
@@ -0,0 +1,94 @@
+In other words, your solution is not just an interpretation of the instructions that "looks right", your solution is a program that _satisfies the given tests_.
+**The tests represent the complete requirements for the exercise.**
+
+## How does Exercism apply TDD?
+
+We've done the work of writing a unit test suite for you.
+Your goal is to write a solution that contains just enough code to make all those unit tests pass.
+
+Keep this in mind: the TDD approach will help you get to the solution, but you don't need to stop there.
+If you want to extend your solution beyond the requirements, you are welcome to do so.
+Should you choose to work with a mentor (and we encourage you to do that once you get the tests passing), they can help you refactor and refine your initial implementation, or even propose new unit tests.
+
+## Working in the online editor
+
+When you're working in the code editor on Exercism's website, you can read the tests but you are not able to edit them.
+All tests will be executed each time you run them, regardless of any "skip" mechanisms noted in the test file.
+
+Test results will default to expanding the first failed test message (although what is displayed varies by track).
+
+Don't be discouraged by a large number of failing tests.
+Focus on making them pass one-by-one.
+
+## Working locally
+
+Many tracks use "skipped" tests in their test files.
+Initially, only the first test is "active" and the remaining are deactivated (how this happens varies by track).
+When you run the test suite in your environment, only the first test runs.
+We do this to encourage you to follow this workflow:
+
+1. Before adding any new code, run the test suite: you should see a failing test.
+1. Add _just enough_ code to pass the test.
+1. Run the test suite.
+1. If the test still fails, repeat step 2.
+1. Once the test passes, refactor your code as desired, ensuring all the activated tests still pass.
+   Refactoring might include:
+    - removing any duplicated code,
+    - spliting long functions into smaller ones
+    - adding comments, etc.
+1. "Unskip" the next test and repeat from step 1.
+
+Repeat these steps until you have unskipped all the tests.
+Once all the tests are passing, congratulations, you have solved the exercise!
+
+Exactly how tests are "unskipped" (or activated) depends on the track.
+For some tracks, it might be commenting or removing an annotation.
+For some tracks, it might be changing an attribute from true to false.
+Take the time to read [the documentation for your track][track-docs]: it will explain these details.
+
+For tracks that don't skip the tests, applying this workflow may be as straightforward as commenting out the tests and uncommenting them one-by-one.
+
+## Rationale for Test-Driven Development
+
+While it may seem like "putting the cart before the horse", there are several good reasons why you might want to write unit tests before writing the implementation code:
```

Comment:
```suggestion
While it may seem like "putting the cart before the horse", there are several good reasons why you might want to write unit tests before writing the implementation code.
```